library/src/main/java/com/felipecsl/gifimageview/library/GifImageView.java

```diff
@@ -151,8 +155,15 @@ public int getGifHeight() {
         }
         gifDecoder.advance();
         try {
-          Thread
-              .sleep(framesDisplayDuration > 0 ? framesDisplayDuration : gifDecoder.getNextDelay());
+          int delay = gifDecoder.getNextDelay();
+          // Sleep for frame duration minus time already spent on frame decode
+          // Actually we need next frame decode duration here,
+          // but I use previous frame time to make code more readable
+          delay -= frame_decode_time;
```

Comment:
You're right. That's why I call Thread.sleep() only if delay is positive. Negative delay means we have already "lost" more time than we had and should display the next frame immediately.

You could also put it this way if you wish:

    if(delay>frameDecodeTime)
      Thread.sleep(delay-frameDecodeTime)

Just don't bother, you can leave it unmerged. You can think about that only if you have complains from other users. Not a critical thing because you'll hardly notice the difference on a powerful device. I have a lot of low-end device users that's why I made this patch for myself.